service/storage/operations.go

```diff
@@ -258,3 +258,67 @@ func (l *Library) LookupSealsForHeight(height uint64, sealIDs *[]flow.Identifier
 func (l *Library) RetrieveResult(txID flow.Identifier, result *flow.TransactionResult) func(*badger.Txn) error {
 	return l.retrieve(encodeKey(prefixResults, txID), result)
 }
+
+// IterateLedger will step through the entire ledger for ledger keys and payloads
+// and call the given callback for each of them.
+func (l *Library) IterateLedger(callback func(path ledger.Path, payload *ledger.Payload) error) func(*badger.Txn) error {
+
+	prefix := encodeKey(prefixPayload)
+	opts := badger.IteratorOptions{
+		PrefetchSize:   100,
+		PrefetchValues: false,
+		Reverse:        true,
+		AllVersions:    false,
+		InternalAccess: false,
+		Prefix:         prefix,
+	}
+	return func(tx *badger.Txn) error {
+
+		it := tx.NewIterator(opts)
+		defer it.Close()
+
+		it.Rewind()
+		for {
+
+			// If we are done iterating, break out.
+			if !it.ValidForPrefix(prefix) {
+				break
+			}
+
+			// First, we extract the path from the key.
+			var path ledger.Path
+			item := it.Item()
+			key := item.Key()
+			copy(path[:], key[1:33])
+
+			// Then we extract the payload from the value.
+			var payload ledger.Payload
+			err := item.Value(func(val []byte) error {
+				err := l.codec.Unmarshal(val, &payload)
+				if err != nil {
+					return err
+				}
+				return nil
+			})
+			if err != nil {
+				return fmt.Errorf("could not decode value (path: %x): %w", path, err)
+			}
```

Comment:
Since item is used in both blocks, it feels clearer to me to extract it right above. WDYT?

```suggestion
			item := it.Item()

			// First, we extract the path from the key.
			var path ledger.Path
			key := item.Key()
			copy(path[:], key[1:33])

			// Then we extract the payload from the value.
			var payload ledger.Payload
			err := item.Value(func(val []byte) error {
				err := l.codec.Unmarshal(val, &payload)
				if err != nil {
					return err
				}
				return nil
			})
			if err != nil {
				return fmt.Errorf("could not decode value (path: %x): %w", path, err)
			}
```

Reply:
I don't see why we need that in its own line.